### DISCUSSION TITLE
Event Driven Architecture

### DISCUSSION THREAD

**Discussion by gradzio** [ngxs/store#1811] [RFC & Proposals]
Wouldnt it be cool if we could dispatch an event at the end of action handler?

This way we could react to it in a layer above and dispatch appropriate action on the store.

One of the reasons why we decided to drop NGXS in our company is that managing events became very difficult.

So what I propose is that we make new type called Event and make @Event() decorator to handle it. It could simply map event payload to action payload and execute same action handler.

What do you guys think?

**Comment by joaqcid** [comment]
hi @gradzio , well you can think of `actions` as events as well, what specifically are you thinking that current functionality doesnt provide for you?

**Reply by gradzio** [reply]
Well if I do, then I will have tight coupling between my modules. Which is exactly what I wanna avoid. In EDA you would have loosely coupled modules. I made a few presentations about it

**Comment by splincode** [comment]
EventEmitter pattern
https://github.com/ngxs-labs/emitter

**Reply by gradzio** [reply]
thanks, but actually I wanna send events between states, not State > Component

**Reply by splincode** [reply]
https://github.com/Angular-RU/angular-ru-sdk/blob/master/packages/ngxs/README.md

is this similar to what you are looking for?

**Reply by gradzio** [reply]
not at all, this is a CRUD implementation. I am talking more about Event Driven Architecture - an enterprise design pattern. Think 300k lines of features, modules, libs and nx monorepo
